### DISCUSSION TITLE
OpenRadioss fatigue calculation

### DISCUSSION THREAD

**Discussion by hdrz** [OpenRadioss/OpenRadioss#2241] [Users]
Hi dear users,
Is there a way to use openradioss for fatigue life estimation?
If not, is there an opensource tool which I can use with openradioss stress results?
Thanks in advance!

**Comment by hdrz** [comment]
Anyone knows?

**Comment by MarianBulla** [comment]
Hi @hdrz,

for fatigue, you usually need the stress in your parts/elements, in the linear-elastic domain. 
Mostly, it is calculated using the implicite method, since in explicite the oscillations are very high (coming from the explicite method).

Then you use these stresses with another tool, which has the info on S-N (stress vs. cycles, usually used in high-loading-frequency domain) or E-N (strain vs. cycles....usually used in in low frequency loading domain). With this additional information (and some more ingrediences), you can estimate the life-time of your product before failure can be expected.

Here, ther are some commercial products available, which can do that (NCode, FemFAT, HyperLife, OptiStruct,....). But you need the Wohler-curve for each of your materials, as additional input.

But you can do the stress calculation with OpenRadioss, for sure.
Here, I would go with OpenRadioss implicite, but explicite should be able to do it, too.

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by hdrz** [comment]
Thank you Marian for the detailed answer. 

I thought this was the case. Essentially the two things needed for fatigue life estimation are stress and material SN/EN curve. For every element I will need to find its material number. 

So my thinking at the moment is to do the computation as a post-processing filter in paraview, probably python filter. I just need to find a way to correlate each element to the correct material. 

Can I find the mat number for each element (or part) in the results file of openradioss?

**Reply by MarianBulla** [reply]
Yes, the material is defined within the /PART definition (the second number).
This points to the material.
The whole part (= all elements of one part) are using the same material. So you need only the S-N curve for that particular material.
And yes, it is mostly done in postprocessing step.
In OpenRadioss, you can store the highest stress within a simulation by using the /FAIL/VISUAL "failure criterion". It will not influece your model, just protocol the stresses (or strains) of one run.
Then you can psotprocess it and compare with your S-N curve.

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by hdrz** [comment]
Thanks for that. I know about the /part keyword (I am a user of radioss and openradioss now) and the material definition. But let me clarify, I am looking for a way to fringe a whole assembly with different materials. 

I am thinking about something on the lines of:  fringe (or filter in paraview) the stress results so that each node, element or integration point will show the local fatigue life. 

I imagine the steps to be as follows:
  1. Prepare the run file as usual, and run it with openradioss. 
  2. Open the results in paraview. Show stress results. 
  3. Prepare a python file with SN/EN curves for each material, amplitudes, number of cycles and whatever else is needed for the computation. Implement the numerics for the computation. 
  4. Use the python file as filter in paraview, taking as inputs for each point the stress and the mat number of that point. 
  5. Enjoy a beautiful fringe, where you can see visually the limiting areas of your design. 
  
For this to work I want to know if each node or element in paraview can be shown with it’s material number that was declared in the /part keyword. (Sorry but I can’t recall this now, will have to check later, or better maybe someone knows if it’s possible?)
